### PR TITLE
fix(project-engine-client): by_tags mock defaults omitted sort_dir to descending (LLMO-6666)

### PR DESCRIPTION
**Pull request by aliciadriani**
## Summary

A **live-verify against prod** (2026-08-04, Lovesac `2840/en`, read-only) found the `by_tags` mock diverged from real Semrush on the **default sort direction**:

- A read with `sort_field=metadata.created_at` and **no `sort_dir`** returned the **exact same order** as an explicit `sort_dir=desc` (`noord == desc`, id-for-id; `asc != desc`). So **live Semrush defaults an omitted `sort_dir` to descending.**
- The mock (added in #1859) defaulted an absent `sort_dir` to **ascending**.
- It's **reachable end-to-end**: the api-service transport sets `sort_dir` only when `order` is present, and the controller's `parsedQuery` doesn't default `order` — so a caller sending `sort` without `order` gets **desc in prod, asc against the mock**.

## Change

- `mock/prompt-sort.js`: ascending only for the literal `'asc'`; **absent / `'desc'` / unrecognised → descending**. Updated the semantics doc and the provenance note.
- `test/mock/prompt-sort.test.js`: assert the descending default; add `'ASC'` case-insensitivity.

## Not changed

- **NULLS position** (missing sort key sorts last, both directions) stays mock-defined — the fleet is fully stamped post-backfill, so there's no live null-metadata prompt to observe. Still matches the spec's "NULLS LAST" (§16 G5); to be confirmed with the Semrush team.
- **No consumer assertion changes**: the LLMO-6667 api-service IT always sends an explicit `order`, so it's unaffected.

Verified locally: unit + e2e green, `tsc` + ESLint clean.

### CODE REVIEW

File: packages/spacecat-shared-project-engine-client/mock/prompt-sort.js [hunk 1]
```diff
@@ -77,7 +78,11 @@ export const sortPromptsByMetadata = (prompts, { sortField, sortDir } = {}) => {
   }
 
   const key = field.slice('metadata.'.length); // 'created_at' | 'updated_at'
-  const descending = String(sortDir ?? '').toLowerCase() === 'desc';
+  // Ascending ONLY for the literal `'asc'`; absent / `'desc'` / unrecognised is descending. Only
+  // the OMITTED default is live-verified (2026-08-04, below); grouping an unrecognised value with
```
Comment (aliciadriani) [inline packages/spacecat-shared-project-engine-client/mock/prompt-sort.js:82]: **Fix — review nit.** Acknowledged in-comment that only the OMITTED-`sort_dir` default (descending) is live-verified; treating an *unrecognised* value the same way is a deterministic mock choice, not verified — and it's unreachable end-to-end anyway (the api-service controller allow-lists `order` to asc/desc, so a bad value 400s before it reaches the mock).

**Review by MysticatBot — APPROVED** [review]
Hey @aliciadriani,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean, focused bug fix with live-verified provenance and matching test updates.
**Complexity: LOW** - small diff, single mock file + test.
_Changes: Corrects the by_tags mock's default sort direction from ascending to descending, matching live-verified Semrush behavior (2 files)._
Note: CI "Test" check is still pending - confirm it passes before merge.

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- nit: No explicit test for `sortDir: null` or `sortDir: undefined` (both hit the `??` fallback and resolve to descending correctly, but an assertion would pin the nullish-coalescing path against future refactors) - `test/mock/prompt-sort.test.js:50`
- nit: No test for `sortDir: ''` (empty string bypasses `??` since it is not nullish, still resolves to descending via `!== 'asc'` - a future simplification to `(sortDir || '').toLowerCase()` could change semantics for this case) - `test/mock/prompt-sort.test.js:50`

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 5s | <b>Cost:</b> $3.31 | <b>Commit:</b> <code>a329db73d1bff529100a6fed70dd89c5290fc985</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: packages/spacecat-shared-project-engine-client/test/mock/prompt-sort.test.js [hunk 2]
```diff
@@ -47,14 +47,30 @@ describe('prompt-sort (by_tags sort_field / sort_dir)', () => {
...
+    // Live-verified 2026-08-04 (prod Lovesac 2840/en): an omitted sort_dir sorts descending.
     expect(ids(sortPromptsByMetadata(prompts, { sortField: 'metadata.created_at' })))
-      .to.deep.equal(['a', 'b', 'c']);
+      .to.deep.equal(['c', 'b', 'a']);
     expect(ids(sortPromptsByMetadata(prompts, { sortField: 'metadata.created_at', sortDir: 'sideways' })))
-      .to.deep.equal(['a', 'b', 'c']);
+      .to.deep.equal(['c', 'b', 'a']);
+  });
+
+  it('resolves null / undefined / empty-string sort_dir to the descending default', () => {
```
Comment (aliciadriani) [inline packages/spacecat-shared-project-engine-client/test/mock/prompt-sort.test.js:58]: **Fix — MysticatBot nits.** Adds explicit assertions that `null` and `undefined` (the `?? ''` nullish branch) and `''` (non-nullish, reaches descending only via `!== 'asc'`) all resolve to the descending default. `''` is asserted separately so a future refactor to `(sortDir || '')` — which would keep `''` descending but change intent — can't slip through silently.

**Review by MysticatBot — APPROVED** [review]
Hey @aliciadriani,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean, focused bug fix with live-verified provenance; new commit addresses both prior nits with well-reasoned test coverage.
**Complexity: LOW** - small diff, single mock + test file.
_Changes: Corrects the by_tags mock's default sort direction from ascending to descending, matching live-verified Semrush behavior, and pins null/undefined/empty-string edge cases with new assertions (2 files)._
Note: CI "Test" and "E2E" checks are still pending - confirm they pass before merge.

## Previously flagged, now resolved
- No explicit test for `sortDir: null` or `sortDir: undefined` - now pinned in dedicated test case with clear comment explaining the `??` vs `||` distinction.
- No test for `sortDir: ''` (empty string) - now asserted in the same test case, guarding against a future refactor to `(sortDir || '')`.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 0m 45s | <b>Cost:</b> $2.59 | <b>Commit:</b> <code>fb8599433a18440636d8ad64d3fadcf28d6fdca0</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
